frontend/src/components/utils.js

```diff
@@ -29,30 +29,41 @@ function removeArrayElement(array, element) {
   array.splice(array.indexOf(element), 1);
 }
 
-function jsonToMermaid(in_json) {
+/**
+ *
+ * @param {*} in_json
+ * @param {string?} highlightedType
+ * @param {string?} highlightedId
+ * @param {string[]} collapsedNodes
+ * @returns
+ */
+function jsonToMermaid(
+  in_json,
+  highlightedType,
+  highlightedId,
+  collapsedNodes,
+) {
   // function to convert the JSON response from a GET request to the /cases/id
   // API endpoint, into the markdown string required for Mermaid to render a flowchart.
   // Nodes in the flowchart will be named [TypeName]_[ID]
   function getNodeName(itemType, itemId) {
     return itemType + "_" + itemId;
   }
 
-  function makeBox(text, shape) {
-    // check if string starts with a number, and if so, prepend a space
-    // to avoid getting a weird unicode character instead
-    if (text.match(/^\d/)) {
-      text = " " + text + " ";
-    }
+  function makeBox(text, shape, name, isCollapsed) {
+    // text is already sanitised at this point, so will not contain <> or "
+
     if (text.length > configData["BOX_NCHAR"]) {
       text = text.substring(0, configData["BOX_NCHAR"] - 3) + "...";
-    } else {
```

Comment:
Is the enforcing of equal width for all boxes, done in the removed block here, now done somewhere else, or no longer done?

Reply:
Now removed - I couldn't find any evidence that it did anything, mermaid ignores trialing spaces when drawing names.